pkg/cluster/docker_desktop.go

```diff
@@ -403,3 +400,61 @@ func (c DockerDesktopClient) settingsForWrite(settings interface{}) interface{}
 
 	return settings
 }
+
+type clientRequest struct {
+	client  HTTPClient
+	method  string
+	url     string
+	headers map[string]string
+	body    []byte
+}
+
+func status2xx(resp *http.Response) bool {
+	return resp.StatusCode >= 200 && resp.StatusCode <= 204
+}
+
+// tryRequest either returns a 2xx response or an error, but not both.
+// If a response is returned, the caller must close its body.
+func (c DockerDesktopClient) tryRequest(label string, creq clientRequest) (*http.Response, error) {
+	klog.V(7).Infof("%s %s\n", creq.method, creq.url)
+
+	body := []byte{}
+	if creq.body != nil {
+		body = creq.body
+		klog.V(8).Infof("Request body: %s\n", string(body))
+	}
+	req, err := http.NewRequest(creq.method, creq.url, bytes.NewReader(body))
+	if err != nil {
+		return nil, errors.Wrap(err, label)
+	}
+
+	for k, v := range creq.headers {
+		req.Header.Add(k, v)
+	}
+
+	resp, err := creq.client.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, label)
+	}
+	if !status2xx(resp) {
+		resp.Body.Close()
+		return nil, errors.Wrap(err, fmt.Sprintf("%s: status code %d", label, resp.StatusCode))
+	}
+
+	return resp, nil
+}
+
+// tryRequests returns the first 2xx response for the given requests, in order,
+// or the error from the last response. If a response is returned, the caller
+// must close its body.
+func (c DockerDesktopClient) tryRequests(label string, requests []clientRequest) (*http.Response, error) {
+	var err error
+	for _, creq := range requests {
+		var resp *http.Response
+		resp, err = c.tryRequest(label, creq)
+		if err == nil {
+			return resp, nil
+		}
+	}
+	return nil, err
```

Comment:
idea: what if we kept track of both errors and returned the one with the "better" response code? (500 as "better"/more informative than 404)